assets/js/blocks/checkout/block.tsx

```diff
@@ -84,22 +84,29 @@ const Checkout = ( {
 		return <CheckoutOrderError />;
 	}
 
+	/**
+	 * Show the login prompt if the user is not logged in
+	 * AND no guest checkout is allowed
+	 * AND no account creation is allowed..
+	 */
 	if (
 		isLoginRequired( customerId ) &&
-		getSetting( 'checkoutAllowsSignup', false )
+		! getSetting( 'checkoutAllowsSignup', false )
 	) {
-		<LoginPrompt />;
+		return <LoginPrompt />;
```

Comment:
Good catch, @mikejolley. I renamed `<LoginPrompt>` to `<MustLoginPrompt>` in this PR. While I could have changed the other `<LoginPrompt>` to `<MayLoginPrompt>`, I decided against it, as this would slightly change the scope of this PR and would require additional testing.